lib/matplotlib/tests/test_widgets.py

```diff
@@ -1019,6 +1019,39 @@ def test_radio_buttons(fig_test, fig_ref):
     ax.text(.25, 1/3, "coffee", transform=ax.transAxes, va="center")
 
 
+@check_figures_equal(extensions=['png'])
+def test_radio_buttons_props(fig_test, fig_ref):
+    label_props = {'color': ['red'], 'fontsize': [24]}
+    radio_props = {'facecolor': 'green', 'edgecolor': 'blue', 'linewidth': 2}
+
+    widgets.RadioButtons(fig_ref.subplots(), ['tea', 'coffee'],
+                         label_props=label_props, radio_props=radio_props)
+
+    cb = widgets.RadioButtons(fig_test.subplots(), ['tea', 'coffee'])
+    cb.set_label_props(label_props)
+    # Setting the label size automatically increases default marker size, so we
+    # need to do that here as well.
+    cb.set_radio_props({**radio_props, 's': (24 / 2)**2})
```

Comment:
I'm a bit worried that this test is checking new API against new API. Because a figure test hasn't been added, I'd advocate for adding a test that sets the properties through lower level API that existed before this PR (getting the artists and setting the properties on them) just to double check everything is working as intended.

Reply:
I can do that, but that would involve accessing private widget details. Not sure how much we like that.

Reply:
I think adding a check of private state is OK here.

Reply:
One point, though, is that the "lower level API that existed before this PR" for the widgets is using the `.circles` etc. which retains fallback behavior that doesn't use scatter at all. The intermediate step between using scatter for this and providing these more public/formal stylization options is not in any released Matplotlib version.

The "old api" and the "new api" don't do the same thing, precisely (at least not to a pixel perfect definition). The point of the change which necessitates this PR was to avoid the transform stack distorting the buttons (as scatter ensures the shape is consistent, but the position is transformed).

I'm not against testing private state, but also think reaching in to private scatter values is just as much "testing new API vs new API" as this is.

Calling `set_radio_props` does not work for the fallback behavior (I don't think it _has_ to, though could see an argument to warn/error if `hasattr(self, "_circles")` to avoid some confusion. That, in turn, would probably have to be suppressed in the `activecolor` setter.)

Reply:
Good point; I think I will add an image test just to be certain about this. I will also add some warnings about the deprecated access.

Reply:
👍 thanks for updating the image test!